treeherder/model/management/commands/cycle_data.py

```diff
@@ -320,6 +321,80 @@ def __lookup_new_signature(self):
             raise LookupError('Exhausted all signatures originating from try repository.')
 
 
+class IrrelevantDataRemoval(RemovalStrategy):
+    """
+    Removes `performance_datum` rows that originate
+    from repositories, other than the ones mentioned
+    in `RELEVANT_REPO_NAMES`, that are more than 6 months old.
+    """
+
+    RELEVANT_REPO_NAMES = [
+        'autoland',
+        'mozilla-central',
+        'mozilla-beta',
+        'fenix',
+        'reference-browser',
+    ]
+
+    def __init__(self, chunk_size: int):
+        self._cycle_interval = timedelta(days=(6 * 30))
+        self._chunk_size = chunk_size
+        self._max_timestamp = datetime.now() - self._cycle_interval
+        self._manager = PerformanceDatum.objects
+        self.__repositories = None
+
+    @property
+    def relevant_repositories(self):
+        if self.__repositories is None:
+            self.__repositories = Repository.objects.filter(
```

Comment:
I see we cache the relevant repository objects. And then we extract only their ids & return them.
We should cache those ids directly.